server/go/payments/payments.go

```diff
@@ -100,3 +122,19 @@ func UpdateShipping(paymentIntent string, r *IntentShippingChangeRequest) (*stri
 
 	return pi, nil
 }
+
+func UpdateCurrencyPaymentMethod(paymentIntent string, r *IntentCurrencyPaymentMethodsChangeRequest) (*stripe.PaymentIntent, error) {
+	currency := r.Currency
+	paymentMethods := r.PaymentMethods
+	
+	params := &stripe.PaymentIntentParams{
+		currency: stripe.String(currency),
+		PaymentMethodTypes: stripe.StringSlice(paymentMethods),
+	}
+	pi, err := paymentintent.Update(paymentIntent, params)
+	if err != nil {
+		return nil, fmt.Errorf("payments: error updating payment intent: %v", err)
+	}
+
+	return pi, nil
+}
```

Comment:
Some of these files are missing a newline at the end.

Reply:
Added new line.